src/wpool_utils.erl

```diff
@@ -28,18 +28,18 @@
 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 
 %% @doc Marks Task as started in this worker
--spec task_init(term(), atom(), infinity | pos_integer()) ->
+-spec task_init(term(), atom(), infinity | pos_integer(), infinity | pos_integer()) ->
```

Comment:
I would use `non_neg_integer()` instead, check the other comment.

Reply:
If we keep the semantic as explained above I'd keep it as a `pos_integer()`